operator/Dockerfile

```diff
@@ -1,29 +1,35 @@
-# Build the adapter binary
+# Build the manager binary
 FROM --platform=$BUILDPLATFORM ghcr.io/kedacore/build-tools:1.19.5 as builder
-
 ARG VERSION=main
 ARG GIT_COMMIT=HEAD
 
 WORKDIR /workspace
-
+# Copy the Go Modules manifests
 COPY go.mod go.mod
 COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
 RUN go mod download
 
-COPY . .
+# Copy the go source
+COPY operator operator
+COPY pkg pkg
+COPY Makefile Makefile
 
 # Build
-# https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
+# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
+# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
+# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
```

Comment:
I don't get what you mean. We weren't setting GOARCH anywhere.

Reply:
This comment came with new Operator SDK scaffolding, I replaced the original Dockerfile and retrofitted our changes. In our case it should read `ARCH` instead of `GOARCH`. I'll fix that.

The comment is just to explaining that by leaving `ARCH` blank, the host architecture will be used. So on Intel Macs it will build amd64 Linux binaries and on Apple Silicon Macs it will build arm64 Linux binaries. It is using Apple as an example because ARM-powered machines are still not mainstreamed on PC-land, but it's also true for Linux/Windows on Intel/AMD vs Linux/Windows on ARM.